x-pack/platform/plugins/shared/dataset_quality/server/rule_types/degraded_docs/register.ts

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { DEFAULT_APP_CATEGORIES } from '@kbn/core/server';
+import { i18n } from '@kbn/i18n';
+import {
+  ALERT_EVALUATION_THRESHOLD,
+  ALERT_EVALUATION_VALUE,
+  ALERT_REASON,
+  DEGRADED_DOCS_RULE_TYPE_ID,
+  STACK_ALERTS_FEATURE_ID,
+} from '@kbn/rule-data-utils';
+import { RuleType } from '@kbn/alerting-plugin/server';
+import {
+  DegradedDocsRuleParams,
+  degradedDocsParamsSchema,
+} from '@kbn/response-ops-rule-params/degraded_docs';
+import { getRuleExecutor } from './executor';
+import {
+  ALERT_EVALUATION_CONDITIONS,
+  ALERT_TITLE,
+  DATASET_QUALITY_AAD_FIELDS,
+  DATASET_QUALITY_REGISTRATION_CONTEXT,
+  DatasetQualityAlert,
+  DatasetQualityAlertContext,
+  DatasetQualityAllowedActionGroups,
+  THRESHOLD_MET_GROUP,
+} from '../types';
+
+export function DegradedDocsRuleType(): RuleType<
+  DegradedDocsRuleParams,
+  never,
+  {},
+  {},
+  DatasetQualityAlertContext,
+  DatasetQualityAllowedActionGroups,
+  never,
+  DatasetQualityAlert
+> {
+  return {
+    id: DEGRADED_DOCS_RULE_TYPE_ID,
+    name: i18n.translate('xpack.datasetQuality.rule.degradedDocs.name', {
+      defaultMessage: 'Degraded docs',
+    }),
+    solution: 'stack',
+    fieldsForAAD: DATASET_QUALITY_AAD_FIELDS,
+    validate: {
+      params: degradedDocsParamsSchema,
+    },
+    schemas: {
+      params: {
+        type: 'config-schema',
+        schema: degradedDocsParamsSchema,
+      },
+    },
+    defaultActionGroupId: THRESHOLD_MET_GROUP.id,
+    actionGroups: [THRESHOLD_MET_GROUP],
+    category: DEFAULT_APP_CATEGORIES.management.id,
+    producer: STACK_ALERTS_FEATURE_ID,
+    minimumLicenseRequired: 'basic',
```

Comment:
Is the licensing looking ok here @LucaWintergerst?

Reply:
license looks good